### ISSUE TITLE
One-off error while readinf ASC files

### ISSUE THREAD

**Issue by ews-ffarella** [GralDispersionModel/GUI#79]
Hi!
I have created a [mesher programm](https://github.com/ewsconsulting/GRAMM_Mesher) for GRAMM based on this repo.
The goal is to use GRAMM in wind resource assessment. 
I have implemented the meshing process in python, to allow for X-Y grid grading (See cell [14] of [this notebook](https://github.com/ewsconsulting/GRAMM_ValidationCases/blob/main/Askervein/manual_meshing_resolution_50m.ipynb)).

During my tests, I have noticed that the terrain heights were not properly interpolated using bilinear interpolation. I have checked this against a pure python implementation, as well as against ArcGisPro.
Basically, I think it comes down to a one-off error.
Unfortunatly, I havent checked the exact line(s) which cause problem, and I have since then updated the code to remove the legacy (Fortran...) 1-based indexing that you are using. 
You should probably have a test case to check your processing of ASC files (I can contribute to that).

Anyways, thanks a lot for the great programm you offer us. It s amazing.

**Comment by MarkusKuntner** [comment]
Thank you very much for the appreciative words. 
Your work is very impressive, I know how difficult it is to read others people's source code.
The terrain rasterization originates entirely from Dietmar Öttl, therefore I have to familiarize myself with this topic first. It is true that in the GUI, but much more in GRAL and GRAMM, the Fortran nomenclature is still used and thus arrays starting with index 1 also occur.
The generated terrain data has been checked by Dietmar, but I am also not sure if the meshing actually works correctly. I don't believe index 1 is the problem (it's probably consistently programmed that way, even if it's no longer well-legible). Maybe the terrain interpolation does not work correctly. I have seen that you have implemented a beautiful and well readable solution for the interpolation. 
If you can help me with the test case but also the solution, I would be very grateful. Aside from that, one should switch to jagged arrays and clean up the code.
I would like to invite you to work in the GUI repo. Are you interested in this?

**Comment by MarkusKuntner** [comment]
Good evening,
I have not yet found the time to look at your work in detail. What I noticed is the interpolation of the land cover data. From our experience, sharp land use boundaries may be necessary, for example, for a lake and its influence on radiation, temperatures, and subsequently on the wind field.

I could reproduce the bug described by you, but I did not search for it in the old code, but programmed the relevant sections with new approaches. If you can find the time to check it out, that would be really great!
The code is in a new branch:
https://github.com/GralDispersionModel/GUI/tree/GrammGrid
Best
Markus
